### ISSUE TITLE
Unable to build master with external XROOTD

### ISSUE THREAD

**Issue by jmcarcell** [root-project/root#12631]
### Describe the bug
When trying to build with an external XRootD, cmake says it finds it but then it says it doesn't and this is making a spack build fail. Relevant lines in the cmake output:
```
-- Looking for XROOTD
-- Found Xrootd version num: 5.5.4.post256 (setting -DROOTXRDVERS=5.5.4.po)
-- Setting OLDPACK TRUE
-- XROOTD not found, enabling 'builtin_xrootd' option
-- Downloading and building XROOTD version 5.5.2
```
When building with XROOTD from spack (where I found this issue) the error is more verbose but I think is the same issue:
```
     121    -- Looking for XROOTD
     122    -- Found Xrootd version num: 5.5.3 (setting -DROOTXRDV
            ERS=500050003)
     123    --              libXrdMain not found: xproofd will be 
            a wrapper around xrootd
     124    --              libXrdClient not found: use built-in
  >> 125    CMake Error at cmake/modules/SearchInstalledSoftware.c
            make:992 (message):
     126      XROOTD not found.  Set environment variable XRDSYS t
            o point to your XROOTD
     127      installation, or include the installation of XROOTD 
            in the
     128      CMAKE_PREFIX_PATH.  Alternatively, you can also enab
            le the option
     129      'builtin_xrootd' to build XROOTD internally
```

### To Reproduce
```
# To install xrootd
git clone https://github.com/xrootd/xrootd
cd xrootd
mkdir build
cd build
cmake .. -DCMAKE_INSTALL_PREFIX=../install
```

```
# To install ROOT
git clone https://github.com/root-project/root
cd root
mkdir builddir
cd builddir
cmake .. -DCMAKE_PREFIX_PATH=/path/to/xrootd/install  -Dbuiltin_xrootd=OFF -Dxrootd=ON -DCMAKE_VERBOSE_MAKEFILE=ON
```

### Setup
cmake 3.26.2
OS: Arch Linux

### Additional context
I'm posting it here since I tried different versions of XRootD and all failed, maybe it's an issue with XROOTD since it's complaining about not finding libXrdMain nor libXrdClient

**Comment by guitargeek** [comment]
Hi @jmcarcell, thanks for opening the issue! I have assigned @bellenot for now, since this issue is related to the build system.

**Comment by bellenot** [comment]
Hi @jmcarcell I managed to build XRootD 5.5.4 and ROOT master using the following XRootD tar file: https://github.com/xrootd/xrootd/archive/refs/tags/v5.5.4.tar.gz, but not when checking it out from Github. Can you try and let us know? And if it works, please ask the XRootD dev team if it's an expected behaviour

**Comment by jmcarcell** [comment]
Hmm I get similar errors with both using the following flags: `-Dbuiltin_xrootd=OFF -Dxrootd=ON -Dfail-on-missing=ON` and having `XRDSYS` pointing to the XRootD installation (or using `CMAKE_PREFIX_PATH`). Which flags did you use for building XRootD and ROOT?

**Comment by bellenot** [comment]
I copied the recipe in your `To Reproduce` section

**Comment by andresailer** [comment]
The related PR broke the LCG builds, probably because we pick up 

https://github.com/root-project/root/blob/master/cmake/modules/FindXROOTD.cmake

cf. https://lcgapp-services.cern.ch/cdash/viewFiles.php?buildid=270695

**Comment by amadio** [comment]
Note that `XrdClient` is the old client from XRootD 4.x, which is already out of support (and EOS 4.x will also go out of support soon). `XrdClient` was removed from the XRootD repository in the XRootD 5.0 release. I think that this is a good time to review the code, drop obsolete things, require at least XRootD 5.x, use `XRootDConfig.cmake` shipped by XRootD, and drop `FindXROOTD.cmake` from ROOT. If `XRootDConfig.cmake` doesn't have something ROOT needs, please let me know and I will fix it for the next patch or feature release, depending on what it is.

**Comment by guitargeek** [comment]
Thank you so much for commenting, @amadio! I Implemented half of what you suggested in this PR:
https://github.com/root-project/root/pull/13752

Feel free to take a look and comment, if you have any suggestions!

**Comment by guitargeek** [comment]
Re-opened because the `XRootDConfig.cmake` from xrootd upstream didn't work after all, see:
https://github.com/root-project/root/pull/14170

**Comment by hahnjo** [comment]
Yes, it appears `XRootDConfig.cmake` is broken in the latest version 5.6.3; I opened https://github.com/xrootd/xrootd/pull/2142 to fix it, then we need a new release and an update in EPEL. Or maybe we can apply the fix on top of 5.6.3 already? @ellert

**Comment by amadio** [comment]
A new release will come this week (5.6.4), I'm already working on it. I think if you can wait a few days it's better.

**Comment by ellert** [comment]
> Yes, it appears `XRootDConfig.cmake` is broken in the latest version 5.6.3; I opened [xrootd/xrootd#2142](https://github.com/xrootd/xrootd/pull/2142) to fix it, then we need a new release and an update in EPEL. Or maybe we can apply the fix on top of 5.6.3 already? @ellert

I have created an update for EPEL (xrootd-5.6.3-3) with this fix and a few others.
https://bodhi.fedoraproject.org/updates/?packages=xrootd
It is currently submitted to EPEL testing. If karma is provided by those who can, it will be pushed to stable faster. Without karma the default time in testing is 7 dats.
